samza-core/src/main/java/org/apache/samza/clustermanager/ContainerProcessManager.java

```diff
@@ -135,23 +169,9 @@ public ContainerProcessManager(Config config, SamzaApplicationState state, Metri
     log.info("Finished container process manager initialization.");
   }
 
-  //for testing
-  ContainerProcessManager(Config config, SamzaApplicationState state, MetricsRegistryMap registry,
-      AbstractContainerAllocator allocator, ClusterResourceManager manager) {
-    this.state = state;
-    this.clusterManagerConfig = new ClusterManagerConfig(config);
-    this.jobConfig = new JobConfig(config);
-    this.hostAffinityEnabled = clusterManagerConfig.getHostAffinityEnabled();
-    this.clusterResourceManager = manager;
-    this.metrics = new ContainerProcessManagerMetrics(config, state, registry);
-    this.containerAllocator = allocator;
-    this.allocatorThread = new Thread(this.containerAllocator, "Container Allocator Thread");
-    this.standbyContainerManager = Optional.empty();
-  }
-
   //package private, used only in tests
   ContainerProcessManager(Config config, SamzaApplicationState state, MetricsRegistryMap registry,
-      ClusterResourceManager resourceManager) {
+      ClusterResourceManager resourceManager, Optional<AbstractContainerAllocator> allocator) {
```

Comment:
Why is the allocator optional? Can a CPM even work without it? Would prefer to keep allocator creation logic in a single place.

Reply:
Because we have tests that pass in their own allocator, and other tests that want the CPM to create the allocator, and we had 3 different constructors for these tests.

Reply:
Oh this is only for testing, it's fine then.